### DISCUSSION TITLE
KIC K8S availability JWT with secret

### DISCUSSION THREAD

**Discussion by liveforfun** [Kong/kubernetes-ingress-controller#2269] [Q&A] [answered]
https://docs.konghq.com/kubernetes-ingress-controller/1.3.x/guides/configure-acl-plugin/

there is a guide using the ACL and JWT Plugins.
in this guide, the user creates their secret for their key.

I just wonder if my system is getting popular and if there are hundreds of millions of people creating JWT token
then can my cluster still be running well?

Maybe I was wrong but I think if the system gets bigger then the secret makes a problem.
is there anyone to think about that?
is there any other way to manage the key?

**Comment by shaneutt** [comment, marked as answer]
> I just wonder if my system is getting popular and if there are hundreds of millions of people creating JWT token
then can my cluster still be running well?
> Maybe I was wrong but I think if the system gets bigger then the secret makes a problem. is there anyone to think about that?

There is certainly some maximum number of credentials `Secrets` which you'll be able to handle, but cluster performance varies from environment to environment. The rule of thumb here is to make sure that you're creating testing infrastructure for your setup and _testing with the load that you expect_ to verify your infrastructure combination appears adequately performant.

>  is there any other way to manage the key?

Technically it is possible to connect directly to the Kong Admin API (this is what the ingress controller utilizes under the hood) which could help you avoid `Secrets` but means you wont be able to use the ingress controller and it's features when doing so.
